ledger-core/insolar/payload/validation.go

```diff
@@ -0,0 +1,178 @@
+		return throw.New("Unordered pending count should be 0")
+	}
+
+	if !m.GetUnorderedPendingEarliestPulse().IsUnknown() {
+		return throw.New("Unordered pending earliest pulse should be Unknown")
+	}
+
+	objectPulse := m.GetAsOf()
+	orderedPendingPulse := m.GetOrderedPendingEarliestPulse()
+
+	if orderedPendingPulse < objectPulse || currPulse < orderedPendingPulse {
+		return throw.New("Incorrect pending ordered pulse number")
+	}
+
+	return nil
+}
+
+func validateReadyStatus(m *VStateReport, currPulse PulseNumber) error {
+	if m.GetUnorderedPendingCount() < 0 || m.GetUnorderedPendingCount() >= 127 {
+		return throw.New("UnorderedPendingCount should be in range [0..127)")
+	}
+
+	if m.GetOrderedPendingCount() < 0 || m.GetOrderedPendingCount() >= 127 {
+		return throw.New("OrderedPendingCount should be in range [0..127)")
+	}
+
+	if m.GetUnorderedPendingEarliestPulse() < 0 || m.GetUnorderedPendingEarliestPulse() > currPulse {
+		return throw.New("UnorderedPendingEarliestPulse should be in range (0..currPulse]")
+	}
+
+	if m.GetOrderedPendingEarliestPulse() < 0 || m.GetOrderedPendingEarliestPulse() > currPulse {
+		return throw.New("OrderedPendingEarliestPulse should be in range (0..currPulse]")
+	}
+
+	return nil
+}
+
+func validateZeroPending(m *VStateReport) error {
+	switch {
+	case m.GetOrderedPendingCount() != 0:
+		return throw.New("OrderedPendingCount should be o")
+	case m.GetUnorderedPendingCount() != 0:
+		return throw.New("UnorderedPendingCount should be o")
+	case !m.GetOrderedPendingEarliestPulse().IsUnknown():
+		return throw.New("OrderedPendingEarliestPulse should be Unknown")
+	case !m.GetUnorderedPendingEarliestPulse().IsUnknown():
+		return throw.New("UnorderedPendingEarliestPulse should be Unknown")
+	}
+
+	return nil
+}
+
+func validateEmptyState(m *VStateReport) error {
+	if m.ProvidedContent == nil {
+		return nil
+	}
+
+	switch content := m.ProvidedContent; {
+	case content.GetLatestValidatedState() != nil:
+		return throw.New("ProvidedContent.LatestValidatedState should be empty")
+	case content.GetLatestValidatedCode() != nil:
+		return throw.New("ProvidedContent.LatestValidatedCode should be empty")
+	case content.GetLatestDirtyState() != nil:
+		return throw.New("ProvidedContent.LatestDirtyState should be empty")
+	case content.GetLatestDirtyCode() != nil:
+		return throw.New("ProvidedContent.LatestDirtyCode should be empty")
+	}
+
+	return nil
+}
+
+func validateEmptyLatest(m *VStateReport) error {
+	switch {
+	case !m.GetLatestValidatedState().IsEmpty():
+		return throw.New("LatestValidatedState should be empty")
+	case !m.GetLatestValidatedCode().IsEmpty():
+		return throw.New("LatestValidatedCode should be empty")
+	case !m.GetLatestDirtyState().IsEmpty():
+		return throw.New("LatestDirtyState should be empty")
+	case !m.GetLatestDirtyCode().IsEmpty():
+		return throw.New("LatestDirtyCode should be empty")
+	}
+
+	return nil
+}
+
+func validateUnimplemented(m *VStateReport) error {
```

Comment:
I'd rather use methods for all message-specific checks, since their name can intersect with another messages validator functions

Reply:
fixed